base/strings/io.jl

```diff
@@ -188,8 +188,8 @@ julia> String(take!(io))
 "Haho"
 ```
 """
-IOBuffer(str::String) = IOBuffer(Vector{UInt8}(str))
-IOBuffer(s::SubString{String}) = IOBuffer(view(Vector{UInt8}(s.string), s.offset + 1 : s.offset + sizeof(s)))
+IOBuffer(str::String) = IOBuffer(unsafe_wrap(Vector{UInt8}, str))
```

Comment:
Probably also `IOBuffer(s::StringBytes) = IOBuffer(s.s)`

Reply:
I have a different question but related to the same line. Why `unsafe_wrap` is used here - for efficiency? I am asking, because it seems that this should work with `StringBytes` equally well.

Reply:
I tried using `StringBytes` and a few things broke that I didn't feel like dealing with. The type `IOBuffer` actually implies that a `Vector{UInt8}` will be used:

```
julia> IOBuffer
Base.GenericIOBuffer{Array{UInt8,1}}
```

so returning a `GenericIOBuffer{StringBytes}` instead can cause surprises.